tools/firtool/firtool.cpp

```diff
@@ -719,36 +602,31 @@ static LogicalResult processBuffer(
 
   // Run the infer-rw pass, which merges read and write ports of a memory with
   // mutually exclusive enables.
-  if (!disableInferRW)
+  if (!disableOptimization)
     pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
         firrtl::createInferReadWritePass());
```

Comment:
I think this should be unconditionally on now. Read-write inference isn't an optimization so much as a defined lowering from CHIRRTL. WDYT?

Reply:
I am not really sure about this one - I believe that the pass is is sort of "best effort" and doesn't really have a spec'ed lowering, and I am pretty sure it is relying on canonicalizations running beforehand to get the best result.  Theoretically  moving this after cross-module optimizations could open up even more port simplifications.

Reply:
I think we might be talking about different passes: this is not the `LowerCHIRRTL` pass, which does have its own "port direction" inference.  This is the one which combines `Read` and `Write` ports into a single `ReadWrite` port.